Add docs (#1)
IntuitiveWebSolutions/gen3-template: docs/README.md

# Gen3 template documentation

- [Project setup](/README.md#project-setup)

> - **Getting started with components**

Reviewer: ```suggestion
- **[Getting started with components](/src/components/README.md)**
```

Author: This is done for a reason. In top level TOC items that are links to a readme, all the sub-items point to a section of the linked readme.

However, in case of components, the sub-items are a mix of both individual readmes as well as sections in readmes. That's why I didn't make it a link to avoid confusion.

>   - [Components directory](/src/components/README.md)

Reviewer: ```suggestion
```
Removing this line to make the top item the Components directory link

Author: See comment above.

  - [Using FontAwesome icons](/src/utilities/README.md#fontawesomejs)
  - [Storing static shared data](/src/utilities/README.md#shareddatajs)
  - [Helper functions](/src/utilities/README.md#helpersjs)
  - [Abstracting Gen2 specific couplings](/src/utilities/README.md#gen2abstractionsjs)
- **Getting started with integration testing**
  - [Writing tests and structuring spec directories and files](/tests/integration/specs/README.md)
  - **[WIP]** [Creating and structuring fixtures](/tests/integration/fixtures/README.md)
  - [Incorporating constants and utilities into your tests](/tests/integration/support/README.md)
    - [Using `testId` constants to target elements in your components](/tests/integration/support/README.md#testid-constants)
    - [Using `elements` constants to target dynamic Design System elements](/tests/integration/support/README.md#elements-constants)
    - [Using `entity` constants to compare values against rendered HTML entities](/tests/integration/support/README.md#entity-constants)
    - [Using utilities to stub page load and gateway responses](/tests/integration/support/README.md#utilsjs)
- **Miscellaneous**

>   - [Helper scripts that bridge the gap between the dev server and the BriteCore instances](/scripts/README.md)

Reviewer: ```suggestion
  - [Running BriteCore as a backend to populate data](/scripts/README.md)
```

  - [Setting default value for the `Get` object when one hasn't been pulled](/src/__mocks__/README.md)

>   - [Customizing HTML template for local development](/public/README.md)

Reviewer: ```suggestion
  - [Using the `public` directory: local and production builds](/public/README.md)
```
